src/data/finset/lattice.lean

```diff
@@ -357,6 +358,28 @@ begin
   refine ⟨max'_mem _ _, λ t Ht, le_antisymm (le_max' s t Ht) (le_trans a (min'_le s t Ht))⟩,
 end
 
+lemma max'_eq_of_dual_min' {s : finset α} (hs : s.nonempty) :
+  max' s hs = of_dual (min' (image to_dual s) (nonempty.image hs to_dual)) :=
```

Comment:
It might also be worth proving `s.max = (image to_dual s).min.map of_dual`

Reply:
If I try
```
lemma to_dual_min_of_dual_eq_max {s : finset α} (hs : s.nonempty) :
s.max = map (image to_dual s).min of_dual
```

Lean says:

```
type mismatch at application
  map (image ⇑to_dual s).min
term
  (image ⇑to_dual s).min
has type
  option (order_dual α) : Type u_1
but is expected to have type
  ?m_1 ↪ ?m_2 : Type (max ? ?)
Additional information:
/home/damiano/Matematica/Lean/mathlib_poly-reverse/src/data/finset/lattice.lean:362:8: context: switched to simple application elaboration procedure because failed to use expected type to elaborate it, error message
  too many arguments
```

I do not know how to deal with ```option```...

Reply:
Your arguments seem to be in the wrong order there.

Reply:
With

```lemma to_dual_min_of_dual_eq_max {s : finset α} (hs : s.nonempty) :
s.max = (image to_dual s).min.map of_dual
```

I get the error:

```
function expected at
  option.map ⇑of_dual (image ⇑to_dual s).min
term has type
  option α
Additional information:
/home/damiano/Matematica/Lean/mathlib_poly-reverse/src/data/finset/lattice.lean:362:29: context: switched to simple application elaboration procedure because failed to use expected type to elaborate it, error message
  too many arguments
```

Reply:
Let's leave this for a future PR.